packages/manager/core/generator/app/conditional-templates/listing/index-api-v2-pagination-cursor.tsx.hbs

```diff
@@ -6,26 +6,51 @@ import { ODS_BUTTON_VARIANT } from '@ovhcloud/ods-components';
 import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
 import { getListingIcebergV2 } from '@/api';
 
-import DatagridWrapper from '@/components/layout-helpers/Listing/DatagridWrapper';
+import TableContainer from '@/components/layout-helpers/Listing/TableContainer';
 import Loading from '@/components/Loading/Loading';
 import ErrorBanner from '@/components/Error/Error';
-import Breadcrumb, {
-} from '@/components/Breadcrumb/Breadcrumb';
-
-
+import Breadcrumb from '@/components/Breadcrumb/Breadcrumb';
 
 export default function Listing() {
   const [pageSize] = useState(10);
-
-  const { data, fetchNextPage, hasNextPage, isError, isLoading, error, }: any = useInfiniteQuery({
+  const {
+    data,
+    fetchNextPage,
+    isError,
+    isLoading,
+    error,
+    isFetching,
+  }: any = useInfiniteQuery({
     initialPageParam: null,
     queryKey: [`servicesListingIceberg`],
-    queryFn: ({ pageParam }) => getListingIcebergV2({ pageSize, cursor: pageParam }),
+    queryFn: ({ pageParam }) =>
+      getListingIcebergV2({ pageSize, cursor: pageParam }),
     staleTime: Infinity,
     getNextPageParam: (lastPage) => lastPage.cursorNext as any,
   });
 
-  const flattenData = data?.pages.map((page: any) => page.data).flat()
+  const tableContainerRef: React.MutableRefObject<HTMLDivElement> = React.useRef<
+    HTMLDivElement
+  >(null);
+
+  const flattenData = React.useMemo(
+    () => data?.pages?.flatMap((page: any) => page.data) ?? [],
+    [data],
+  );
+  const totalFetched = flattenData.length;
+
+  const fetchMoreOnBottomReached = React.useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 500px of the bottom of the table, fetch more data if we can
+        if (scrollHeight - scrollTop - clientHeight < 500 && !isFetching) {
+          fetchNextPage();
+        }
+      }
+    },
+    [fetchNextPage, isFetching, totalFetched],
+  );
```

Comment:
**nitpick:** Cannot hard-code 500px. Please use dynamic values like `vh` or similar properties and evaluate if the user has reached the end of datagrid.

Reply:
we had to define a size height of the datagrid only for the scroll infinite, we can define a constant for configure it.
But the offset scroll in the datagrid is only in PX